### PR TITLE
add initial commit

### PR DESCRIPTION
**Pull request by jondmcelroy**
add LoggerBootstrapper for initializing service fabric project logging
add Options for turning on the application insights or activity logging
implement an example project with example keysets, actiontype and extensions
write up the readme file with instructions on how to use it

### CODE REVIEW

File: examples/ActivityLogger/Program.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using SoCreate.Extensions.Logging.ActivityLogger;
+
+namespace ActivityLogger
+{
+    internal class Program
+    {
+        public static void Main(string[] args)
+        {
+            LoggerBootstrapper.InitializeServiceFabricRegistration(
+                (serviceName, configuration, addloggingToServiceContext) =>
```
Comment (lurock) [inline examples/ActivityLogger/Program.cs]: Rename parameter from "addloggingToServiceContext" to "addServiceContextToLogging".

File: src/SoCreate.Extensions.Logging/ApplicationInsightsLogger/ApplicationInsightsLoggerExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            var telemetry = GetTelemetry(logEvent, formatProvider);
+
+            // Add Operation Id
+            if (Activity.Current?.RootId != null)
+            {
+                telemetry.Context.Operation.Id = Activity.Current?.RootId;
+            }
+
+            // TODO add User ID
```
[the remark this answers is not in the document]
Comment (jondmcelroy) [reply src/SoCreate.Extensions.Logging/ApplicationInsightsLogger/ApplicationInsightsLoggerExtensions.cs]: We need the accountService/ Usercontext object concept to be created.

File: src/SoCreate.Extensions.Logging/ActivityLogger/ActivityLogger.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        private readonly string _version;
+
+        public ActivityLogger(ILogger logger, IConfiguration configuration)
+        {
+            _logger = logger;
+            _activityLogType = configuration.GetValue<string>("ActivityLogger:ActivityLogType") ?? "DefaultType";
+            _version = configuration.GetValue<string>("ActivityLogger:ActivityLogVersion") ?? "v1";
+        }
+
+        public void LogActivity<TActivityEnum>(IActivityKeySet keySet, TActivityEnum actionType, string message,
```
[the remark this answers is not in the document]
Comment (jondmcelroy) [reply src/SoCreate.Extensions.Logging/ActivityLogger/ActivityLogger.cs]: Should I just add it to the interface? Why have it be an extension method? Do you think it is cleaner that way?

File: src/SoCreate.Extensions.Logging/LoggerBootstrapper.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+using Serilog.Events;
+using SoCreate.Extensions.Logging.ApplicationInsightsLogger;
+using SoCreate.Extensions.Logging.ServiceFabric;
+using SerilogILogger = Serilog.ILogger;
+
+namespace SoCreate.Extensions.Logging
+{
+    public static class LoggerBootstrapper
+    {
+        public static LoggingLevelSwitch LoggingLevelSwitch { get; set; }
```
[the remark this answers is not in the document]
Comment (jondmcelroy) [reply src/SoCreate.Extensions.Logging/LoggerBootstrapper.cs]: I added some documentation on it. Each project will need to implement either a developer UI or endpoint to set that value.

File: src/SoCreate.Extensions.Logging/LoggerBootstrapper.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            if (propertyName == null) throw new ArgumentNullException($"{propertyName}");
+            var scalar = new ScalarValue(scalarValue);
+            return e =>
+            {
+                if (!e.Properties.TryGetValue(propertyName, out var propertyValue)) return false;
+                
+                if (propertyValue is StructureValue stValue)
+                {
+                    var value = stValue.Properties.FirstOrDefault(cc => cc.Name == "Id");
```
[the remark this answers is not in the document]
Comment (jondmcelroy) [reply src/SoCreate.Extensions.Logging/LoggerBootstrapper.cs]: Left over from the event id. removed

File: examples/ActivityLogger/Program.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                {
+                    var host = new HostBuilder()
+                        .ConfigureHostConfiguration(configHost => { configHost.AddConfiguration(configuration); })
+                        .ConfigureServices((hostContext, services) =>
+                        {
+                            services.AddActivityLogger(typeof(ActivityLogger<>));
+                            services.AddSingleton(configuration);
```
[the remark this answers is not in the document]
Comment (jondmcelroy) [reply examples/ActivityLogger/Program.cs]: Not needed it appears.

**Pull request merged**